src/App.tsx

```diff
@@ -1,14 +1,50 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [filterBy, setFilterBy] = useState<string>('all');
+  const [isLoaded, setIsLoaded] = useState<boolean>(false);
+  const [query, setQuery] = useState<string>('');
+
+  useEffect(() => {
+    getTodos()
+      .then(result => {
+        setTodos(result);
+        setIsLoaded(true);
+      });
+  }, []);
+
+  const lowerCompare = (str: string) => (
+    str.toLowerCase().includes(query.toLowerCase())
+  );
+
+  const filterTodos = todos
+    .filter(todoItem => {
+      if (filterBy === 'active') {
+        return !todoItem.completed;
+      }
+
+      if (filterBy === 'completed') {
+        return todoItem.completed;
+      }
+
+      return todoItem;
+    })
+    .filter(todoItem => (
+      lowerCompare(todoItem.title)
+    ));
```

Comment:
would be good to use `useMemo`  the performance of your task will be much better (*optional)